### PR TITLE
Migrate remaining 'publish panel' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.

PR migrates remaining `publish-panel.test.js` e2e tests to Playwright.

## Why?
See #38851.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/various/publish-panel.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/various/publish-panel.spec.js [hunk 1]
```diff
@@ -57,4 +57,60 @@ test.describe( 'Post publish panel', () => {
...
+		await page
+			.getByRole( 'region', 'Editor top bar' )
+			.getByRole( 'button', { name: 'Publish', exact: true } )
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Editor publish' } )
+				.locator( ':focus' )
+		).toHaveText( 'Publish' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/publish-panel.spec.js:77]: Could we instead rewrite this into `toBeFocused` to better match the description?

File: test/e2e/specs/editor/various/publish-panel.spec.js [hunk 2]
```diff
@@ -57,4 +57,60 @@ test.describe( 'Post publish panel', () => {
...
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'Test Post' );
+		await editor.publishPost();
+
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Editor publish' } )
+				.locator( ':focus' )
+		).toHaveText( 'Test Post' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/publish-panel.spec.js:93]: Ditto.

**Review by kevin940726 — APPROVED** [review]
Thanks! Just small nitpicks that don't block the PR.

File: test/e2e/specs/editor/various/publish-panel.spec.js (hunk 1 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/publish-panel.spec.js:77]: Don't have a strong opinion, but the original test was checking the `activeElement`.

File: test/e2e/specs/editor/various/publish-panel.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/publish-panel.spec.js:93]: Same here.

**Pull request merged**